patterns/grok-patterns

```diff
@@ -35,7 +35,7 @@ HOSTPORT %{IPORHOST}:%{POSINT}
 
 # paths
 PATH (?:%{UNIXPATH}|%{WINPATH})
-UNIXPATH (/([\w_%!$@:.,+~-]+|\\.)*)+
+UNIXPATH (?>\/[[\w_%!$@:.,-]+|\\.]*)+
```

Comment:
@purbon I can explain more. In regexp, the brackets `[]` and parentheses `()` mean completely different things. Parentheses are syntax for grouping. Brackets are syntax for character classes.

These two things mean completely different things: `[abcd]` vs `(abcd)`. The first (brackets) matches a single character that can be either a, b, c, or d. The second (parentheses) matches exactly the text `abcd`.

So my concern about the `()` to `[]` change you made:

```diff
-[\w_%!$@:.,+~-]+|\\.
+[[\w_%!$@:.,-]+|\\.]
```

The way I read this regexp, the former (removed code) means basically `one or more of the following characters: word character, `_`, `%`, .... `-`` _OR_ a literal `.` character.

The second (new code) means basically: `match one single character that can be [some of those characters]`

Example:

```
# Old code, matching a single part of the path.
>> /[\w_%!$@:.,+~-]+|\\./.match("/hello/world")
=> #<MatchData "hello">

# New code,
>> /[[\w_%!$@:.,-]+|\\.]/.match("/hello/world")
=> #<MatchData "h">
```

The character class is changed in this PR by adding, i think, left bracket, pipe, and backslash.

This PR description looks like it is supposed to fix the behavioral bug (ReDOS), not add new things that can match. Right?